### ISSUE TITLE
Lazy computation for layout attributes

### ISSUE THREAD

**Issue by mikehouse** [iwheelbuy/SquareMosaicLayout#8]
When collection view has thousands of items then method **prepare()** works slow that causes some lags/freezes, especially it is noticeable on device orientation change. Thanks.

**Comment by iwheelbuy** [comment]
Yep, that is the problem. The layout was not designed to work with such a great amount of items...
